sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -5306,6 +5306,40 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val EXCEPT_OPEREATOR_OUTPUT_AS_LAZY_VAL = buildConf("spark.sql.exceptOperatorOutputAsLazyVal")
+    .internal()
+      .doc("When set to true, Except operator's output will be a lazy val. This is a " +
+        "performance for querires with a large number of stacked EXCEPTs. This is because of " +
+        "rules like WidenSetOperationTypes that traverse the logical plan tree and call output " +
+        "on each Except node. Such traversal has quadratic complexity: " +
+        "O(number_of_nodes * (1 + 2 + 3  + ... + number_of_nodes)).")
+    .version("4.0.0")
+    .booleanConf
+    .createWithDefault(true)
+
+  val INTERSECT_OPEREATOR_OUTPUT_AS_LAZY_VAL =
+    buildConf("spark.sql.intersectOperatorOutputAsLazyVal")
+      .internal()
+      .doc("When set to true, Intersect operator's output will be a lazy val. This is a " +
+        "performance for querires with a large number of stacked INTERSECTs. This is because of " +
+        "rules like WidenSetOperationTypes that traverse the logical plan tree and call output " +
+        "on each Intersect node. Such traversal has quadratic complexity: " +
+        "O(number_of_nodes * (1 + 2 + 3  + ... + number_of_nodes)).")
+      .version("4.0.0")
+      .booleanConf
+      .createWithDefault(true)
+
+  val UNION_OPEREATOR_OUTPUT_AS_LAZY_VAL = buildConf("spark.sql.unionOperatorOutputAsLazyVal")
```

Comment:
Don't think need too fine granularity, how about to introduce at least one config for all operators:
```suggestion
  val LAZY_SET_OPERATOR_OUTPUT = buildConf("spark.sql.lazySetOperatorOutput.enabled")
```

Reply:
Done.